examples/text_to_image/train_text_to_image.py

```diff
@@ -691,6 +710,40 @@ def collate_fn(examples):
                         accelerator.save_state(save_path)
                         logger.info(f"Saved state to {save_path}")
 
+                        if args.validation_prompt:
+                            pipeline = StableDiffusionPipeline.from_pretrained(
+                                args.pretrained_model_name_or_path,
+                                text_encoder=text_encoder,
+                                vae=vae,
+                                unet=unet,
+                                revision=args.revision,
+                            )
+                            pipeline = pipeline.to(accelerator.device)
+                            pipeline.set_progress_bar_config(disable=True)
+
+                            # run inference
+                            prompt = [args.validation_prompt]
+                            with torch.autocast("cuda"):
+                                images = pipeline(prompt, num_images_per_prompt=args.num_validation_images).images
+
+                            for i, image in enumerate(images):
+                                image.save(os.path.join(args.output_dir, f"sample-{global_step}-{i}.jpg"))
+                            for tracker in accelerator.trackers:
+                                if tracker.name == "tensorboard":
+                                    np_images = np.stack([np.asarray(img) for img in images])
+                                    tracker.writer.add_images("validation", np_images, epoch, dataformats="NHWC")
+                                if tracker.name == "wandb":
+                                    tracker.log(
+                                        {
+                                            "validation": [
+                                                wandb.Image(image, caption=f"{i}: {args.validation_prompt}")
+                                                for i, image in enumerate(images)
+                                            ]
+                                        }
+                                    )
+                            del pipeline
+                            torch.cuda.empty_cache()
+
             logs = {"step_loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0]}
```

Comment:
Do we also want to add another final inference logging as done in the `train_dreambooth_lora.py` example? 

https://github.com/huggingface/diffusers/blob/4fc7084875ac672980c4e0207fa325c8c418a88e/examples/dreambooth/train_dreambooth_lora.py#L983